Dockerfile.ci

```diff
@@ -11,7 +11,7 @@ RUN curl -sL https://git.io/goreleaser -o /tmp/goreleaser.sh \
 
 # Cache optimization
 COPY go.mod go.sum Makefile ./
-RUN go mod download && make vendor
+RUN go mod download
```

Comment:
should we replace /w `make tidy`?

Reply:
make tidy ends up running `go mod tidy` and the thing with go mod tidy is that it updates and then, downloads the modules listed in go.mod. The risk with that is that say, in runtime, `make tidy`/`go mod tidy` ends up upgrading one of our modules to a newer major breaking version, that would break our code and it would be hard to track it coz in the repository everything would seem to be working fine.

For reference, this is a similar concern with addon-operator. If you run `go mod tidy` over it, it will upgrade logr/go-logger from 0.4.0 to 1.2.0 which breaks the entire addon-operator.

although as of now, we don't have any module in go.mod, which even upgraded, would cause any breakages in runtime. So, we can afford to proceed with `make tidy` as of now but god knows what will happen in the future 🤞🏼 

wdyt :)

Reply:
awesome let's keep it out.